src/components/renders/bootstrap/switch/switch-render.tsx

```diff
@@ -0,0 +1,50 @@
+import { EventEmitter } from "@stencil/core";
+type Constructor<T> = new (...args: any[]) => T;
+export function SwitchRender<T extends Constructor<{}>>(Base: T) {
+  return class extends Base {
+    caption: string;
+    checked: boolean;
+    disabled = false; // This takes its value form HTML
+    element: HTMLElement;
+    id: string;
+    private inputId: string;
+
+    onChange: EventEmitter;
+    private getValueFromEvent(event: UIEvent): boolean {
+      return event.target && (event.target as HTMLInputElement).checked; // Return if HTMLInput is checked or not
+    }
+    handleChange(event: UIEvent) {
+      this.checked = this.getValueFromEvent(event); // Check if input element is checked in each click
+      this.onChange.emit(event);
+    }
+    render() {
+      if (!this.inputId) {
+        // Allways run because is undefined, and !undefined = true
+        this.inputId = this.id
+          ? `${this.id}_checkbox`
+          : `gx-checkbox-auto-id-${autoCheckBoxId++}`;
+      }
+      // Attributes for input. Values taked form HTML and asigned here
+      const inputAttrs = {
+        "aria-checked": this.checked ? "true" : undefined,
+        "aria-disabled": this.disabled ? "true" : undefined,
+        checked: this.checked,
+        class: "switch",
+        disabled: this.disabled,
+        id: this.inputId,
+        onChange: this.handleChange.bind(this),
+        type: "checkbox"
+      };
+      return (
+        <div class="form-group">
```

Comment:
The `<div>` element is unneeded. You can remove it and the end result will be the same. It will be added when used inside a gx-form-field element.